src/oci-umount.c

```diff
@@ -630,21 +635,30 @@ int main(int argc, char *argv[])
 	}
 	int target_pid = YAJL_GET_INTEGER(v_pid);
 
-	/* OCI hooks set target_pid to 0 on poststop, as the container process already
-	   exited.  If target_pid is bigger than 0 then it is the prestart hook.  */
-	if ((argc > 2 && !strcmp("prestart", argv[1])) || target_pid) {
-		ret = parseBundle(&node, &config_mounts, &config_mounts_len);
+	/* OCI hooks set target_pid to 0 on poststop, as the container process 
+	   already exited.  If target_pid is bigger than 0 then it is a start 
+	   hook. 
+	   In most cases the calling program should pass in a argv[1] option,
+	   like prestart, poststart or poststop.  In certain cases we also
+	   support passing of no argv[1], and then default to prestart if the 
+	   target_pid != 0, poststop if target_pid == 0.
+	*/
+	if ((argc >= 2 && !strcmp("prestart", argv[1])) ||	if ((argc >= 2 && !strcmp("prestart", argv[1])) ||
+	    (argc == 1 && target_pid)) {
```

Comment:
Is there something wrong with above if condition? I think you have copied pasted one line extra.